### PR TITLE
Mom6 Add check for obs_too_deep

### LINKED ISSUE #490: bug: MOM6 obs too deep
:bug: 🐘 Your bug may already be reported!
Please search on the [issue tracker](https://github.com/NCAR/DART/issues) before creating a new issue.

### Describe the bug

There is no catch after the find layer loop to check whether the observation is too deep.

https://github.com/NCAR/DART/blob/502af7865e258530cba7c7daff6aed4390cbb649/models/MOM6/model_mod.f90#L286-L328

There is an OBS_TOO_DEEP status, but it is never used:
https://github.com/NCAR/DART/blob/502af7865e258530cba7c7daff6aed4390cbb649/models/MOM6/model_mod.f90#LL120C22-L120C35

1. List the steps someone needs to take to reproduce the bug.  
MOM6 with WOD data base obs
3 ensemble member CESM run here:
/glade/scratch/hkershaw/c.T62_g16.Alper.ens3/run
note the ensemble members are identical.
2. What was the expected outcome?
Correct forward operator result and status
3. What actually happened?  
Obs too deep not caught.

### Error Message  
Please provide any error messages.
You get errors trying to access incorrect memory, e.g.
 PE 0:  filter trace: Before computing prior observation values
```
mlx5: r5i3n3: got completion with error:
00000000 00000000 00000000 00000000
00000000 00000000 00000000 00000000
000000e9 00000000 00000000 00000000
00000000 00008813 100036c4 03421ad2
```
or

```
forrtl: severe (408): fort: (3): Subscript #1 of the array PE_TO_TASK_LIST has value -278 which is less than the lower bound of 1

Image              PC                Routine            Line        Source             
filter             0000000000BEB1FF  Unknown               Unknown  Unknown
filter             0000000000649C66  ensemble_manager_         966  ensemble_manager_mod.f90
filter             00000000005CBDDB  distributed_state         104  distributed_state_mod.f90
filter             00000000005CB635  distributed_state          79  distributed_state_mod.f90
filter             00000000005CAE90  distributed_state          54  distributed_state_mod.f90
filter             000000000066FECF  model_mod_mp_mode         342  model_mod.f90
filter             0000000000475E2F  obs_def_mod_mp_ge        1342  obs_def_mod.f90
filter             00000000004167C0  forward_operator_         447  forward_operator_mod.f90
filter             000000000040EAB1  forward_operator_         283  forward_operator_mod.f90
filter             00000000004E6152  filter_mod_mp_fil         864  filter_mod.f90
filter             00000000005ECC73  MAIN__                     20  filter.f90
```

### Which model(s) are you working with?
MOM6

### Screenshots   
Screen Shot 2023-06-07 at 10 14 01 AM

The obs depth is greater than the total depth. 
Note I think the total depth might be incorrect (edit ~~will update later~~ see comment below), but still the obs_too_deep should be caught.

### Version of DART      
Which version of DART are you using? 
You can find the version using `git describe --tags`  
v10.7.2-2-g5f3235bbc

### Have you modified the DART code?   
Yes/No  
If your code changes are available on GitHub, please provide the repository.
https://github.com/NCAR/DART/tree/mom6-scripting

### Build information
Please describe:  
 1. Cheyenne
 2. intel

### PR DESCRIPTION
**Pull request by hkershaw-brown**
## Description:

* Added error check for observations deeper than the model.
* Added sensible defaults in work/input.nml for running MOM6 with CESM
   Particularly single_file_in/out = .false. If this is true filter does some wacky things to the MOM6 restart files. 

### Fixes issue

Fixes  #490

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Documentation changes needed?

- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly.

### Tests
Please describe any tests you ran to verify your changes.
Running CESM-MOM6 with DART
/glade/scratch/hkershaw/c.T62_g16.Alper.ens3/run

## Checklist for merging

- [ ] Updated changelog entry
- [ ] Documentation updated
- [ ] Update conf.py

## Checklist for release
- [ ] Merge into main
- [ ] Create release from the main branch with appropriate tag
- [ ] Delete feature-branch

## Testing Datasets

- [x] Dataset needed for testing available upon request
- [ ] Dataset download instructions included
- [ ] No dataset needed

### CODE REVIEW

**Review by mjs2369 — COMMENTED** [review]
@hkershaw-brown what's the motivation behind the changes to the nml? Are these related to the obs_too_deep catch? Or just the nml values you used for your test case?

If the latter should they be in this PR?

**Comment by hkershaw-brown** [comment]
> @hkershaw-brown what's the motivation behind the changes to the nml? Are these related to the obs_too_deep catch? Or just the nml values you used for your test case?
> 
> If the latter should they be in this PR?

The nml changes are more sensible defaults for MOM6
the single_file_in/out = .true. that was there  caused some problems
   `single_file_in = .true.` filter errors out because it is expecting a members directory
   `single_file_out = .true.` no error, but filter overwrites the existing restart files, a bunch of MOM6 info gets removed, MOM6 can't restart.

`c.e22.GMOM.T62_g16.nuopc.001` is a case specific name, I switched it to a generic name `mom6.static.nc`

`assimilate_these_obs_types` is a list of the obs from the World Ocean Data base

`init_time_days/seconds = -1` use the model time

I'd like to keep the nml changes in this PR, since they are needed for running DART with CESM-MOM6 and real obs.

The `trace execution` and `ens_size`changes  can be removed.

File: models/MOM6/work/input.nml [hunk 1]
```diff
@@ -44,7 +44,7 @@
    output_sd                    = .true.
    write_all_stages_at_end      = .false.
 
-   ens_size                     = 20,
+   ens_size                     = 3,
```
Comment (hkershaw-brown) [inline models/MOM6/work/input.nml:47]:
```suggestion
   ens_size                     = 20,
```

File: models/MOM6/work/input.nml [hunk 2]
```diff
@@ -75,7 +75,7 @@
    inf_sd_lower_bound          = 0.0,                     0.0,
    inf_sd_max_change           = 1.05,                    1.05,
 
-   trace_execution              = .false.,
+   trace_execution              = .true.,
```
Comment (hkershaw-brown) [inline models/MOM6/work/input.nml:78]:
```suggestion
   trace_execution              = .false.,
```

**Review by mjs2369 — APPROVED** [review]
Ready to merge with main.

**Pull request merged**